### PR TITLE
LIBS-27 Adds a RefreshTokenValidator

### PR DESCRIPTION
**Pull request by JeremyGibson**
Modifies the token validators so that authn needs don't clobber app posts.

### CODE REVIEW

File: auth_checker/models/models.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class AuthnTokenRequestBody(BaseModel):
     """
 
     token: str
-    authn_type: AuthNTypes
+    authn_type: AuthNTypes = AuthNTypes.OAUTH2
```
Comment (ryan-semmler) [inline auth_checker/models/models.py:37]: Do we want this hardcoded to OAUTH2?

**Review by lsanchez44 — APPROVED** [review]
all of this looks above board

File: auth_checker/models/models.py [hunk 2]
```diff
@@ -129,10 +129,41 @@ def render(self):
...
+            raise HTTPException(
+                400, detail="Token has an invalid signature. Check the JWT_SECRET variable."
+            )
+
+
+class RefreshTokenValidator(BaseTokenValidator):
+    def __init__(self, body: AuthnTokenRequestBody):
+        super().__init__()
+        if body:
+            self.token = body.token
```
Comment (ryan-semmler) [inline auth_checker/models/models.py:160]: Looks like `body` can't ever be falsey here, we can maybe skip the `if` statement

**Review by ryan-semmler — APPROVED** [review]
lgtm, just a couple thoughts

**Pull request merged**